src/ORM/Connect/Query.php

```diff
@@ -3,12 +3,18 @@
 namespace SilverStripe\ORM\Connect;
 
 use SilverStripe\Core\Convert;
-use Iterator;
 use Traversable;
 
 /**
- * Abstract query-result class. A query result provides an iterator that returns a map for each record of a query
- * result.
+ * Abstract query-result class. A query result provides an IteratorAggregate that returns a map for each record of a
+ * query result.
+ *
+ * This should be subclassed by an actual database implementation. It will only
+ * ever be constructed by a subclass of Database. The result of a database query - an iteratable object
+ * that's returned by DB::query(), Database::query() or Database::preparedQuery().
```

Comment:
Might be better to just say "returned by various low-level ORM methods.", to avoid this list becoming stale.

Reply:
Good suggestion. Done.